### PR TITLE
Merge RotWing State Machine

### PR DESCRIPTION
**Pull request by tmldeponti**
I have modified the new state machine by @fvantienen to be independent of guidance indi hybrid. It is then possible to also use this state machine with the oneloop controller. This allows us to merge the two state machines and have only one common shared one.

### CODE REVIEW

**Comment by dewagter** [comment]
I agree to move some functions to maximize reusability, but can you please fix the compile issue. (```make tudelft``` and  ```make tests```)

**Comment by tmldeponti** [comment]
> I agree to move some functions to maximize reusability of modules, but can you please fix the compile issue. (`make tudelft` and `make tests`)

Yes I am on it! :)

**Comment by tmldeponti** [comment]
> I agree to move some functions to maximize reusability of modules, but can you please fix the compile issue. (`make tudelft` and `make tests`)

I fixed all errors in the test modules. As for test tudelft, the airframes with the oneloop controller will be fixed in the next PR I will open about the oneloop.

File: conf/airframes/tudelft/rotwing_v3c_oneloop_optitrack_ext_pose.xml [hunk 1]
```diff
@@ -216,7 +214,24 @@
...
+        <define name="QUAD_MAX_AIRSPEED"        value="12.0"/>  <!-- Maximum quadrotor airspeed (with pusher motor)-->
+        <define name="QUAD_MAX_DECELERATION"    value="0.75"/>  <!-- Maximum horizontal deceleration in quad mode -->
+        <define name="SKEW_UP_AIRSPEED"         value="10.0"/>  <!-- Airspeed where the skewing starts when going up in airspeed -->
+        <define name="SKEW_DOWN_AIRSPEED"       value="8.0"/>   <!-- Airspeed where the skewing starts when going down in airspeed -->
 
+        <define name="SKEW_REF_MODEL"           value="TRUE"/>  <!-- Enable second order reference model for the skewing command -->
+        <define name="SKEW_REF_MODEL_P_GAIN"    value="0.001"/> <!-- Skewing reference model proportional gain -->
+        <define name="SKEW_REF_MODEL_D_GAIN"    value="0.003"/> <!-- Skewing reference model differential gain -->
+        <define name="SKEW_REF_MODEL_MAX_SPEED" value="20"/>    <!-- Maximum rotational skewing speed bound for the reference model -->
+        <define name="ROTWING_FW_SKEW_ANGLE"    value="85"/>
```
Comment (fvantienen) [inline conf/airframes/tudelft/rotwing_v3c_oneloop_optitrack_ext_pose.xml:233]: That define does not work

File: conf/airframes/tudelft/rotwing_v3c_oneloop_simulation.xml [hunk 2]
```diff
@@ -186,7 +182,24 @@
...
+        <define name="QUAD_MAX_AIRSPEED"        value="12.0"/>  <!-- Maximum quadrotor airspeed (with pusher motor)-->
+        <define name="QUAD_MAX_DECELERATION"    value="0.75"/>  <!-- Maximum horizontal deceleration in quad mode -->
+        <define name="SKEW_UP_AIRSPEED"         value="10.0"/>  <!-- Airspeed where the skewing starts when going up in airspeed -->
+        <define name="SKEW_DOWN_AIRSPEED"       value="8.0"/>   <!-- Airspeed where the skewing starts when going down in airspeed -->
 
+        <define name="SKEW_REF_MODEL"           value="TRUE"/>  <!-- Enable second order reference model for the skewing command -->
+        <define name="SKEW_REF_MODEL_P_GAIN"    value="0.001"/> <!-- Skewing reference model proportional gain -->
+        <define name="SKEW_REF_MODEL_D_GAIN"    value="0.003"/> <!-- Skewing reference model differential gain -->
+        <define name="SKEW_REF_MODEL_MAX_SPEED" value="20"/>    <!-- Maximum rotational skewing speed bound for the reference model -->
+        <define name="ROTWING_FW_SKEW_ANGLE"    value="85"/>
```
Comment (fvantienen) [inline conf/airframes/tudelft/rotwing_v3c_oneloop_simulation.xml:201]: Same here

File: conf/airframes/tudelft/rotwing_v3g_oneloop_optitrack_ext_pose.xml [hunk 3]
```diff
@@ -0,0 +1,373 @@
...
+        <define name="QUAD_MAX_AIRSPEED"        value="12.0"/>  <!-- Maximum quadrotor airspeed (with pusher motor)-->
+        <define name="QUAD_MAX_DECELERATION"    value="0.75"/>  <!-- Maximum horizontal deceleration in quad mode -->
+        <define name="SKEW_UP_AIRSPEED"         value="10.0"/>  <!-- Airspeed where the skewing starts when going up in airspeed -->
+        <define name="SKEW_DOWN_AIRSPEED"       value="8.0"/>   <!-- Airspeed where the skewing starts when going down in airspeed -->
+
+        <define name="SKEW_REF_MODEL"           value="TRUE"/>  <!-- Enable second order reference model for the skewing command -->
+        <define name="SKEW_REF_MODEL_P_GAIN"    value="0.001"/> <!-- Skewing reference model proportional gain -->
+        <define name="SKEW_REF_MODEL_D_GAIN"    value="0.003"/> <!-- Skewing reference model differential gain -->
+        <define name="SKEW_REF_MODEL_MAX_SPEED" value="20"/>    <!-- Maximum rotational skewing speed bound for the reference model -->
+        <define name="ROTWING_FW_SKEW_ANGLE"    value="85"/>
```
Comment (fvantienen) [inline conf/airframes/tudelft/rotwing_v3g_oneloop_optitrack_ext_pose.xml:233]: Same

**Review by fvantienen — CHANGES_REQUESTED** [review]

**Review by fvantienen — APPROVED** [review]

**Pull request merged**